components/cdc/src/endpoint.rs

```diff
@@ -677,6 +679,9 @@ impl<T: 'static + RaftStoreRouter<E>, E: KvEngine> Endpoint<T, E> {
         let checkpoint_ts = request.checkpoint_ts;
         let sched = self.scheduler.clone();
 
+        let is_build_resolver =
+            kv_api == ChangeDataRequestKvApi::TiDb && !delegate.has_tidb_downstream();
```

Comment:
How about named as `has_resolver()` ? And just set true here.
To build resolver if `NOT has_tidb_downstream` is wired.